lib/Graph/Graph.cpp

```diff
@@ -2258,6 +2258,24 @@ static bool isGraphStorageNode(Node *n, const Function *F) {
   return false;
 }
 
+/// \returns true if Node \p node is written into, either as a result, or as an
+/// overwritten input.
+static bool hasWriters(const Node *node) {
+  for (auto &U : node->getUsers()) {
+    auto *N = U.getUser();
+    // See if node is used as a result anywhere.
+    for (unsigned i = 0; i < N->getNumResults(); i++)
+      if (node == N->getNthResult(i))
+        return true;
+    // See if node is used as an overwritten input anywhere.
+    for (unsigned i = 0; i < N->getNumInputs(); i++)
+      if (N->isOverwrittenNthInput(i))
+        if (node == N->getNthInput(i))
+          return true;
+  }
+  return false;
+}
```

Comment:
Instead of adding this method, I believe we could modify the loop that check the placeholders and fail when we see the first use.
I am referring to the loop that start with:
 // Make sure all the placeholders are at most written once.

We could remove the:
      if (!isa<Constant>(input)) {
        continue;
      }

and adapt and the following assert:
 if (varToFirstDef != placeholderWrittenTo.end()) {